e2e/e2e.sh

```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
 
 # Run the pytest command
+source venv/bin/activate
```

Comment:
This has strict requirements on the current working directory. Either document it or use `$0`  to locate it relative to the command itself.